src/battle_script_commands.c

```diff
@@ -1725,9 +1725,7 @@ u32 GetTotalAccuracy(u32 battlerAtk, u32 battlerDef, u32 move, u32 atkAbility, u
     if (gFieldStatuses & STATUS_FIELD_GRAVITY)
         calc = (calc * 5) / 3; // 1.66 Gravity acc boost
 
-    // With high affection/friendship there's a chance to evade a move by substracting 10% of its accuracy.
-    // I can't find exact information about that chance, so I'm just gonna write it as a 20% chance for now.
-    if (B_AFFECTION_MECHANICS == TRUE && GetBattlerFriendshipScore(battlerDef) >= FRIENDSHIP_150_TO_199 && (Random() % 100) <= 20)
+    if (B_AFFECTION_MECHANICS == TRUE && GetBattlerAffectionHearts(battlerDef) == AFFECTION_FIVE_HEARTS)
         calc = (calc * 90) / 100;
```

Comment:
This shouldn't be happening always. It really *is* based on a chance. [Source](https://bulbapedia.bulbagarden.net/wiki/Affection).

Reply:
Bulbapedia says it always substracts 10% from accuracy though? I dont read anywhere that says its a chance.

Reply:
> Bulbapedia says it always substracts 10% from accuracy though?

It does not. It's specifically stated that it happens "sometimes".

> At affection level 4 and higher, a Pokémon can sometimes:

>    [Avoid attacks](https://bulbapedia.bulbagarden.net/wiki/Stat#Evasion), by subtracting 10% from their accuracy.

Reply:
Oh yeah, the `if` statement needs to be updated if you intend to submit the PR again down the road, since it uses `AFFECTION_FIVE_HEARTS` instead of `AFFECTION_FOUR_HEARTS`.

Reply:
Yea, it can sometimes avoid attacks since the accuracy is lowered. Which happens all the time.

Reply:
> Oh yeah, the `if` statement needs to be updated if you intend to submit the PR again down the road, since it uses `AFFECTION_FIVE_HEARTS` instead of `AFFECTION_FOUR_HEARTS`.

I specifically use SWSH and SV's mechanics. They are detailed on the friendship page, and they are slightly different.

Reply:
That's not really what it says though.

It says that at Affection Level 4 or higher there's a chance that a reduction of the incoming attack's accuracy happens, it does not say that there's a permanent decrease in accuracy.